issue-5230

LionOnTheChase opened the pull request:
added traceid and spanid to decision log.

test for checking traceid in decision log is not passing

ashutosh-narkar commented on internal/internal.go:
```diff
@@ -519,6 +521,22 @@ func (p *envoyExtAuthzGrpcServer) log(ctx context.Context, input interface{}, re
+		info.SpanID = sctx.SpanID().String()
+	}
+
+	p.manager.Logger().WithFields(map[string]interface{}{
```
We should remove this log as much of this is already logged at debug level in the `check` call. This is also resulting in the test failure.

ashutosh-narkar commented on internal/internal_test.go:
```diff
@@ -165,6 +167,38 @@ func TestCheckAllow(t *testing.T) {
 	}
 }
 
+func TestCheckTraceIDInAllowedRequest(t *testing.T) {
```
You could use something like [this](https://github.com/open-policy-agent/opa/blob/main/test/e2e/distributedtracing/distributedtracing_test.go#L148) as a reference to setup the unit test.

ashutosh-narkar reviewed (commented):
Thanks for working on this @LionOnTheChase. Few comments inline.

ashutosh-narkar commented:
Closing this for now. Feel free to pick this back up and address the comments when you get a chance. Thanks for looking into this.

LionOnTheChase commented:
@ashutosh-narkar  , thank you for your patience. I have addressed your comments and created a new PR https://github.com/open-policy-agent/opa-envoy-plugin/pull/462 . 

Kindly review
